huaweicloud/services/dms/resource_huaweicloud_dms_kafka_smart_connect_task.go

```diff
@@ -0,0 +1,382 @@
+			},
+			"destination_file_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the destination file type of the smart connect task.`,
+			},
+			"record_delimiter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: `Specifies the record delimiter of the smart connect task.`,
+			},
+			"topics": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				Description:  `Specifies the topics of the task.`,
+				ExactlyOneOf: []string{"obs_destination_descriptor.0.topics", "obs_destination_descriptor.0.topics_regex"},
+			},
+			"topics_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				Description:  `Specifies the topics regular expression of the smart connect task.`,
+				ExactlyOneOf: []string{"obs_destination_descriptor.0.topics", "obs_destination_descriptor.0.topics_regex"},
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceDmsKafkaSmartConnectTaskCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	// createKafkaSmartConnectTask: create DMS kafka smart connect task
+	var (
+		createKafkaSmartConnectTaskHttpUrl = "v2/{project_id}/connectors/{connector_id}/sink-tasks"
+		createKafkaSmartConnectTaskProduct = "dms"
+	)
+	createKafkaSmartConnectTaskClient, err := cfg.NewServiceClient(createKafkaSmartConnectTaskProduct, region)
+
+	if err != nil {
+		return diag.Errorf("error creating DMS Client: %s", err)
+	}
+
+	connectorID := d.Get("connector_id").(string)
+	createKafkaSmartConnectTaskPath := createKafkaSmartConnectTaskClient.Endpoint + createKafkaSmartConnectTaskHttpUrl
+	createKafkaSmartConnectTaskPath = strings.ReplaceAll(createKafkaSmartConnectTaskPath, "{project_id}",
+		createKafkaSmartConnectTaskClient.ProjectID)
+	createKafkaSmartConnectTaskPath = strings.ReplaceAll(createKafkaSmartConnectTaskPath, "{connector_id}", connectorID)
+
+	createKafkaSmartConnectTaskOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	createKafkaSmartConnectTaskOpt.JSONBody = utils.RemoveNil(buildCreateKafkaSmartConnectTaskBodyParams(d))
+
+	createKafkaSmartConnectTaskResp, createErr := createKafkaSmartConnectTaskClient.Request("POST",
+		createKafkaSmartConnectTaskPath, &createKafkaSmartConnectTaskOpt)
+
+	if createErr != nil {
+		return diag.Errorf("error creating DMS kafka smart connect task: %v", createErr)
+	}
+
+	kafkaSmartConnectTaskRespBody, err := utils.FlattenResponse(createKafkaSmartConnectTaskResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	taskID := utils.PathSearch("task_id", kafkaSmartConnectTaskRespBody, nil)
+	if taskID == nil {
+		return diag.Errorf("error retrieving DMS kafka smart connect task id: %v", taskID)
```

Comment:
it's unnecessary to print the value of `taskID ` in the error message as it ls already **nil**.

Reply:
done